com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs

```diff
@@ -1313,17 +1317,11 @@ internal void HandleApproval(ulong clientId, bool createPlayerObject, ulong? pla
                 // This packet is unreliable, but if it gets through it should provide a much better sync than the potentially huge approval message.
                 SyncTime();
 
-
-                if (createPlayerObject)
-                {
-                    var networkObject = NetworkSpawnManager.CreateLocalNetworkObject(false, 0, playerPrefabHash ?? NetworkConfig.PlayerPrefabHash.Value, null, position, rotation);
-                    NetworkSpawnManager.SpawnNetworkObjectLocally(networkObject, NetworkSpawnManager.GetNetworkObjectId(), false, true, clientId, null, false, 0, false, false);
-
-                    ConnectedClients[clientId].PlayerObject = networkObject;
-                }
-
+                //Reset our observed objects list
+                //[NSS]: This assures we have an up to date list of all observed objects
                 m_ObservedObjects.Clear();
```

Comment:
I wonder why m_ObservedObjects is not a local to HandleApproval?  It's only used in HA, and only in the approved state, and in that state, it is unconditionally cleared, then built up and read from.